main.tf

```diff
@@ -0,0 +1,196 @@
+#
+# IAM resources
+#
+data "aws_iam_policy_document" "ecs_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "ecs_service_role" {
+  name               = "ecs${var.environment}ServiceRole"
+  assume_role_policy = "${data.aws_iam_policy_document.ecs_assume_role.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_service_role" {
+  role       = "${aws_iam_role.ecs_service_role.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceRole"
+}
+
+data "aws_iam_policy_document" "ecs_autoscale_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["application-autoscaling.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "ecs_autoscale_role" {
+  name               = "ecs${var.environment}AutoscaleRole"
+  assume_role_policy = "${data.aws_iam_policy_document.ecs_autoscale_assume_role.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_service_autoscaling_role" {
+  role       = "${aws_iam_role.ecs_autoscale_role.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceAutoscaleRole"
+}
+
+#
+# Security group resources
+#
+resource "aws_security_group" "main" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Name        = "sg${var.name}LoadBalancer"
+    Project     = "${var.project}"
+    Environment = "${var.environment}"
+  }
+}
+
+#
+# ALB resources
+#
+resource "aws_alb" "main" {
+  security_groups = ["${aws_security_group.main.id}"]
+  subnets         = ["${var.public_subnet_ids}"]
+  name            = "alb${var.environment}${var.name}"
+
+  access_logs {
+    bucket = "${var.access_log_bucket}"
+    prefix = "${var.access_log_prefix}"
+  }
+
+  tags {
+    Name        = "alb${var.environment}${var.name}"
+    Project     = "${var.project}"
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_alb_target_group" "main" {
+  name = "tg${var.environment}${var.name}"
```

Comment:
It won't, but I think it is worth mentioning what convention is expected for `name` (`CamelCase` without spaces) in the `README`.